app/src/components/SoldArtworkList.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import SoldArtworkItem from './SoldArtworkItem';
+import CardColumns from 'react-bootstrap/CardColumns';
+import { ContractProps } from '../helper/eth';
+
+interface SoldArtworkListState {
+  balance: number;
+  tokenIds: Array<number>;
+  prices: Array<number>;
+  newOwners: Array<string>;
+}
+
+class SoldArtworkList extends React.Component<ContractProps, SoldArtworkListState> {
+  constructor (props: ContractProps) {
+    super(props);
+    this.state = {
+      balance: 0,
+      tokenIds: [],
+      prices: [],
+      newOwners: [],
+    };
+  }
+
+  componentDidMount (): void {
+    this.loadSales();
+  }
+
+  shouldComponentUpdate (): boolean {
+    this.loadSales();
+    return true;
+  }
+
+  async loadSales (): Promise<void> {
+    const governance = this.props.contracts.Governance;
+    const currentAccount = this.props.accounts[0];
+    let eventsFound = 0;
+    const tokensArray: any[] = [];
+    const priceArray: any[] = [];
+    const newOwnerArray: any[] = [];
+
+    await governance.getPastEvents(
```

Comment:
When I'm testing this locally it doesn't seem to give me all past sales if I make multiple sales (only the most recent sale, which also disappears after a while). Not sure if this is just on my end.

Reply:
Will test more robustly: it may just be that that’s how long events are retrievable for but that seems wrong

Reply:
This does indeed seem to be the case: our events don't seem to be well logged when we emit them / don't persist for long

Reply:
Seems like the answer should be to just filter from block 0, yet the filter functionality is the functionality that seems to be broken

Reply:
Probably should create a task for it to see if anyone else can figure it out